[improve JENKINS-30107] print ip causing the failed login

lexxxel opened the pull request:
this PR addresses the problem described [here](https://issues.jenkins-ci.org/browse/JENKINS-30107).
By adding the ip address to the log, I'm able to use fail2ban to prevent brute force attacks, but of cause someone could also write a plugin for this.

akshayabd committed:
[JENKINS-26580] Initial implementation of JNLP3-connect protocol

akshayabd committed:
[JENKINS-26580] Updated implementation of Jnlp3 protocol

pjanouse committed:
[JENKINS-31847] CLI command clear-queue extracted from core to CLI

clear-queue covered by test-cases

Dohbedoh committed:
[JENKINS-32134]: Improved the Build Other Projects help message for projects inside Folder.

Dohbedoh committed:
[JENKINS-32134]: Referenced CloudBees Folder Plugin and Multi-Branch Project Plugin

Dohbedoh committed:
[JENKINS-32134]: Used the core terminology "ItemGroup". Simplified and added examples. Added the closing paragraph tags.

kohsuke committed:
Moved some of the server-side logic to remoting

This way, the inner-working of the protocol can be kept private within
the remoting module, which increases a modularity.

kohsuke committed:
JnlpAgentReceiver needs to be adjusted for the new type

Given that there's only one caller, I think handling AbstractMethodError
is better than doing the compatibility method ping pong

kohsuke committed:
Merge remote-tracking branch 'origin/master' into jnlp3-protocol

Conflicts:
	core/src/main/java/jenkins/slaves/JnlpAgentReceiver.java
	pom.xml

kohsuke committed:
Fixing a compilation problem

kohsuke committed:
Java7 code transformation

kohsuke committed:
Log cannot be opened twice so it needs to be kept around.

kohsuke committed:
Terminology fix

pjanouse committed:
[JENKINS-32684] Improved test coverage of CLI command 'console'

Dohbedoh committed:
[JENKINS-32134]: Change the terminology to make it more understandable for users.

pjanouse committed:
[JENKINS-33037] Fixed malformed ranges in RangeSet.fromString()

Fix for rejecting malformed ranges in
hudson.model.Fingerprint.RangeSet.fromString(...)

ydubreuil committed:
[JENKINS-33038] correctly set executor in FutureImpl

This patch fixes cancellation of the build linked to the Future
object. Before, the executor set in FutureImpl was always empty,
because WorkUnitContext.createWorkUnit() is not called from an
Executor thread anymore.

kohsuke committed:
2.54 contains JNLP3 change

varmenise committed:
[JENKINS-32831] if metadata for tool installer does not exist should not fail

varmenise committed:
fix typo in method's signature

kohsuke committed:
Need to call ChannelConfigurator

kohsuke committed:
When this gets merged let's record the version number

russinholi committed:
Fix proxy server name documentation (#2060)

jglick committed:
We expect a modelClass to be assignable to ModelObject.

jglick committed:
COMPUTER-scoped permissions are in fact often checked on Node.

jglick committed:
Computer.CREATE should be scoped to JENKINS, not COMPUTER, since it is checked on Jenkins, not a Node or Computer.
Compare Item.CREATE, scoped to ITEM_GROUP, not ITEM.

kohsuke committed:
A/B test this feature in production

JNLP3 is activated now for 10% of users. Let's keep it like this for a
while and if no major issue occurs we should expose it to everyone.

kohsuke committed:
Merge remote-tracking branch 'origin/master' into jnlp3

Conflicts:
	pom.xml

kohsuke committed:
Pick up a new version of remoting

See https://github.com/jenkinsci/remoting/pull/78

lexxxel committed:
add client ip in case of failed login

lexxxel committed:
add client ip in case of failed login

lexxxel committed:
add client ip in case of failed login part 2

lexxxel committed:
add client ip in case of failed login part 2

lexxxel committed:
fix build problems

lexxxel committed:
fix build problems

Dohbedoh committed:
[JENKINS-32525] - Handle null @AncestorInPath when selecting Downstream Jobs.

kohsuke committed:
Pick up a released version

stephenc committed:
Align open(URL) with getInputStream(URL) so that they both use the proxy settings valid at the point of call

stephenc committed:
Jenkins.getInstance() is safe to assume non-null

- Only ever called after extensions are on the classpath and before termination

stephenc committed:
Jenkins.getInstance() is safe to assume non-null (x1)

- This code path is only ever called after extensions are on the classpath and before termination
- The other code path may just possibly be called when `Jenkins.theInstance == null`, for clarification I assert that this can never happen, but as all it does is log the errors in a claimed failed startup I will leave this with just a TODO marker for now

stephenc committed:
Jenkins.getInstance() is safe to assume non-null

- Only ever called after extensions are on the classpath while performing initialization and during termination before `Jenkins.theInstance` is set to `null`

stephenc committed:
Noting that there is a >99% certainty that these case use getInstance()

stephenc committed:
Jenkins.getInstance() is safe to assume non-null

- Only ever called after extensions are on the classpath and before termination

stephenc committed:
Noting that there is a >99% certainty that these case use getInstance()

stephenc committed:
Noting that there is a >99% certainty that these case use getInstance()

stephenc committed:
Jenkins.getInstance() is safe to assume non-null

- Only ever called after extensions are on the classpath and before termination

stephenc committed:
Noting that there is a potentially valid argument for allowing this method to work without the singleton...

- But the type of tests here would be very restricted in what they could safely do anyway, so they would largely be restricted to verifying data migration during xml loading

stephenc committed:
Noting that there is a >99% certainty that these can use getInstance()

- Plugins using these methods will be safe
- The code path I need to analyse is the `Jenkins.super()` constructor code path. As Jenkins extends AbstractCIBase if there is a call there that requires the queue lock during construction it may require these methods to be safe against being called before the singleton has been put in place

stephenc committed:
Jenkins.getInstance() is safe to assume non-null

- Only ever called after extensions are on the classpath and before termination

stephenc committed:
Noting that this usage required JENKINS-33377 before it can be fixed

stephenc committed:
Jenkins.getInstance() is safe to assume non-null

- Only ever called after extensions are on the classpath and before termination

stephenc committed:
Arguably all these fireXXX methods should no-op if the extensions are not available

stephenc committed:
Jenkins.getInstance() is safe to assume non-null

- Only ever called after extensions are on the classpath and before termination
- Noting that the one valid check for null should actually be using a guard on extensions being available and not a check for null

stephenc committed:
It would be better if we had an explicit method to identify when code is run in a thread on the master vs in a thread on a slave

 - because forcing the slave to load the Jenkins class into its classloader is probably not a good plan

stephenc committed:
Jenkins.getInstance() is safe to assume non-null

- Only ever called after extensions are on the classpath and before termination
- Only ever called from a thread on the master

stephenc committed:
It would be better if we had an explicit method to identify when code is run in a thread on the master vs in a thread on a slave

 - because forcing the slave to load the Jenkins class into its classloader is probably not a good plan

stephenc committed:
Fix javadoc comment

stephenc committed:
Jenkins.getInstance() is safe to assume non-null

- Only ever called after extensions are on the classpath and before termination

stephenc committed:
Jenkins.getInstance() is safe to assume non-null

- Only ever called after extensions are on the classpath and before termination

stephenc committed:
Should have been using `ExtensionList.lookup(Class)` anyway

stephenc committed:
Jenkins.getInstance() is safe to assume non-null

- Only ever called after extensions are on the classpath and before termination
- If it were somehow or other called after `Jenkins.theInstance == null` then at that point the queue has already been persisted so the trigger could have no effect anyway... much better to surface in the shutdown logs about the missed trigger than either return false or incorrectly report the job as triggered

stephenc committed:
Jenkins.getInstance() is safe to assume non-null

- Only ever called after extensions are on the classpath and before termination

stephenc committed:
Jenkins.getInstance() is safe to assume non-null

- Only ever called after extensions are on the classpath and before termination

stephenc committed:
[FIXED JENKINS-33384] clean up the PluginServletFilter instances on Jenkins.cleanUp()

jtnord committed:
[FIXED JENKINS-33395] install an uncaught exception handler

stephenc committed:
Jenkins.getInstance() will be @Nonnull in Jenkins 2.0+

- Code that is running from a plugin and on the master's JVM is guaranteed to never get null from this method (any cases where you do get null are bugs in core)
- Code that is running from a plugin and on a remote JVM should never be allowed to load the Jenkins class in their classloader, so should never use Jenkins.getInstance()... we are annotating the method with @Nullable so that such code can have some evolution time
- Code that is running in core and on one of two special paths should use the Jenkins.getInstanceOrNull() method so that the UI can be presented to users before the singleton has been instantiated / after the singleton has been destroyed
- The remaining 95% of uses in core (and 100% of uses in plugins) can safely assume that the instance is never null

lexxxel committed:
raise the log level and improve the output message

lexxxel committed:
raise the log level and improve the output message

stephenc committed:
[FIXED JENKINS-33414] Added an API to allow identification of the Jenkins JVM

- Fix the ProxyConfiguration code to not require Jenkins class to be loaded in remote JVMs

stephenc committed:
[JENKINS-33414] Fix Javadoc comments

stephenc committed:
[JENKINS-33414] Add a test case that verifies for real

daniel-beck committed:
Merge pull request #2093 from stephenc/jenkins-33384

[FIXED JENKINS-33384] clean up the PluginServletFilter instances on Jenkins.cleanUp()

stephenc committed:
Merge pull request #2100 from stephenc/jenkins-33414

[FIXED JENKINS-33414] Added an API to allow identification of the Jenkins JVM

jtnord commented on core/src/main/java/jenkins/security/SecurityListener.java:
```diff
@@ -112,10 +112,10 @@ public static void fireLoggedIn(@Nonnull String username) {
     /** @since 1.569 */
-    public static void fireFailedToLogIn(@Nonnull String username) {
-        LOGGER.log(Level.FINE, "failed to log in: {0}", username);
+    public static void fireFailedToLogIn(@Nonnull String username, @Nonnull String ip) {
```
this breaks backwards compatability.

jtnord commented:
unfortunately the changes as implemented breaks backwards compatibility as all security Listeners (can be defined in plugins) will need to be updated.

jtnord commented on core/src/main/java/jenkins/security/SecurityListener.java:
```diff
@@ -71,7 +71,7 @@
      * @param username the user
      */
-    protected abstract void failedToLogIn(@Nonnull String username);
+    protected abstract void failedToLogIn(@Nonnull String username, @Nonnull String ip);
```
this breaks backwards compatability for all `SecurityListener` implementations.

daniel-beck commented on core/src/main/java/hudson/security/AuthenticationProcessingFilter2.java:
```diff
@@ -103,7 +103,12 @@ protected void onUnsuccessfulAuthentication(HttpServletRequest request, HttpServ
         if (auth != null) {
-            SecurityListener.fireFailedToLogIn(auth.getName());
+        	//is client behind something?
+			String ipAddress = request.getHeader("X-FORWARDED-FOR");  
```
This should be taken care of by the server. We don't use this for the `Cause.RemoteCause` either, for example:
https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/jenkins/model/ParameterizedJobMixIn.java#L258

See https://github.com/jenkinsci/winstone/pull/13

stephenc committed:
[FIXED JENKINS-33425] Refactored the cleanUp() method to log progress and push for completion

stephenc committed:
Merge remote-tracking branch 'jenkins/master' into alt-pr-2090

stephenc committed:
[JENKINS-33414] Ensure that the tests do not leave the JVM marker flag set

stephenc committed:
[JENKINS-33414] Cover one final theoretical race condition

jtnord committed:
Add a BIG FAT WARNING for security managers.

kohsuke committed:
Merge pull request #2010 from jenkinsci/jnlp3

[FIXED JENKINS-26580] Activate JNLP3 support

stephenc committed:
Merge pull request #2104 from stephenc/jenkins-33414-followup

[JENKINS-33414] Ensure that the tests do not leave the JVM marker flag set

stephenc committed:
[JENKINS-33425] w00t we can use addSuppressed

stephenc committed:
Merge remote-tracking branch 'jenkins/master' into jenkins-33425

stephenc committed:
[JENKINS-33425] Better timer cancellation

kohsuke committed:
Revert "[JENKINS-32190] Make UsageStatistics#isDue check Jenkins init level before generating stats."

This reverts commit d8a13ac07c4357eb74ab45a03955c0db430e3aa9.

I noticed a strange drop in usage statistics from the mid Jan 2016,
which corresponds to the time this change was released in the wild,
and further drop when LTS 1.642.2 is released with this change at
the end of Feb.

I spent some time trying to determine how this might cause instances
to stop sending data, but I couldn't come up with any. I left those
notes in JENKINS-32190. There was some strange behaviours, but none
explains what this is.

So I'd like to back out this change and see if that makes any impact
on the stats collection. If it backs up, then we know empirically
this change is to blame, so that would justify spending further
effort or come up with a fix from a different angle.

stephenc committed:
Merge pull request #2103 from stephenc/jenkins-33425

[FIXED JENKINS-33425] Refactored the cleanUp() method to log progress and push for completion

kohsuke committed:
Merge pull request #2109 from jenkinsci/JENKINS-32190

Revert JENKINS-32190 fix

stephenc committed:
Merge remote-tracking branch 'jenkins/master' into alt-pr-2090

stephenc committed:
Merge pull request #2094 from stephenc/alt-pr-2090

Fix Jenkins.getInstance() to make life easier for plugin authors

stephenc committed:
Merge remote-tracking branch 'jenkins/master' into bb7c8fc-followup

stephenc committed:
This one we should leave as is

stephenc committed:
The `@Initializer` and `@Terminator` annotated methods are only executed where `Jenkins.getInstance()!=null`

stephenc committed:
These could be null in case of concurrent calls to restart

fbelzunc committed:
Improve logging and error message when JNLP is already in use

stephenc committed:
Merge pull request #2092 from stephenc/bb7c8fc-followup

Use getInstance() in place of getInstanceOrNull() for execution paths that cannot have a null

jglick committed:
Simplifying equals/hashCode using java.util.Objects.

jglick committed:
[FIXED JENKINS-33467] Do not store redundant copies of Cause in CauseAction.

daspilker committed:
updated windows-package-checker to 1.2 to avoid compile time warnings

JENKINS-32978

daniel-beck committed:
Merge pull request #2114 from jglick/CauseAction-JENKINS-33467

[JENKINS-33467] Do not store duplicate Cause’s

jtnord committed:
fixt typo

fix typos spotted by @jglick's eagle eyes

jglick committed:
[JENKINS-33467] Noting merge of #2114.

Dohbedoh committed:
[JENKINS-32525] - Added warning.

qerub committed:
Fix typos in Swedish translation

jtnord committed:
Merge pull request #2119 from qerub/fix-typos-in-swedish-translation

Fix typos in Swedish translation

kohsuke committed:
[maven-release-plugin] prepare release jenkins-1.653

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

kohsuke committed:
updated changelog for release

daniel-beck committed:
Merge pull request #2120 from daspilker/JENKINS-32978-windows-package-checker

[JENKINS-32978] updated windows-package-checker to 1.2 to avoid compile time warnings

daniel-beck committed:
Merge pull request #2118 from pjanouse/JENKINS-32684

[JENKINS-32684] Improved test coverage of CLI command 'console'

kohsuke committed:
[FIXED JENKINS-21484]

Don't require ancient .NET 2.0 runtime. .NET 4.0 can host this
executable, too.

See https://github.com/kohsuke/winsw#net-runtime-40

kohsuke committed:
Merge pull request #2124 from kohsuke/jenkins-21484

[FIXED JENKINS-21484]

damianszczepanik committed:
Remove mention of http.proxyHost property

Issue #2060

damianszczepanik committed:
Merge pull request #2126 from damianszczepanik/proxy

Remove mention of http.proxyHost property

olivergondza committed:
Merge pull request #1942 from pjanouse/JENKINS-31847

[JENKINS-31847] CLI command clear-queue extracted from core to CLI

olivergondza committed:
Merge pull request #2112 from fbelzunc/JENKINS-33453

[FIXED JENKINS-33453] Improve logging and error message when JNLP is already in use

olivergondza committed:
Fix changelog links

olivergondza committed:
Noting #2112

olivergondza committed:
Merge pull request #1958 from Dohbedoh/JENKINS-32134

[FIXED JENKINS-32134]: Improved the Build Other Projects help message for p…

olivergondza committed:
Merge pull request #2064 from russinholi/master

Fix proxy server name documentation (#2060)

olivergondza committed:
Merge pull request #2043 from ydubreuil/fix-futureimpl

[FIXED JENKINS-33038] correctly set executor in FutureImpl

olivergondza committed:
Merge pull request #2049 from pjanouse/JENKINS-33037

[FIXED JENKINS-33037] Fixed malformed ranges in RangeSet.fromString()

olivergondza committed:
Note merges

jglick committed:
[JENKINS-33467] Clarifying that CauseAction.getCauses is immutable and you should construct the action with the causes you want.

jglick committed:
Merge pull request #2065 from jglick/PermssionScope.COMPUTER

PermissionScope clarifications

jglick committed:
Setting font-size does in fact work here.

jglick committed:
[FIXED JENKINS-18032] Crumbs must be appended when using post=true requiresConfirmation=true.

daniel-beck committed:
Add a title for the confusing 'Use browser' option

daniel-beck committed:
Merge pull request #2131 from jglick/requiresConfirmation-post-context-menu-JENKINS-18032

[JENKINS-18032] Fix Delete Project from context menu when using CSRF defense

daniel-beck committed:
Merge pull request #2050 from varmenise/JENKINS-32831

[FIXED JENKINS-32831] do not fail if there are no metadata for tool installers

oleg-nenashev committed:
Merge pull request #2095 from jtnord/jenkins-33395

[FIXED JENKINS-33395] install an uncaught exception handler

jglick committed:
[FIXED JENKINS-32525] Merged #2087.

jglick committed:
Merge pull request #2130 from jglick/bad-style

Set a smaller font on metadata in log

olivergondza committed:
Merge pull request #2132 from daniel-beck/use-browser-title

Add a title for the confusing 'Use browser' option

batmat committed:
[INFRA-588] Upgrade&Remove some inherited values

See
https://github.com/jenkinsci/pom/commit/4518f1577489d7a7fd91c0213ab708defb77a67f

effective-pom diff between & after this change:

    $ diff before.xml after.xml
    7c7
    <     <version>1.35</version>
    ---
    >     <version>1.36</version>
    85c85
    <       <url>http://maven.jenkins-ci.org:8081/content/repositories/releases</url>
    ---
    >       <url>https://repo.jenkins-ci.org/releases/</url>
    89c89
    <       <url>http://maven.jenkins-ci.org:8081/content/repositories/snapshots</url>
    ---
    >       <url>https://repo.jenkins-ci.org/snapshots/</url>

jglick committed:
Merge pull request #2143 from batmat/INFRA-588

[INFRA-588] Upgrade + Remove inherited values

stephenc committed:
[FIXED JENKINS-33681] Plugin filters were failing to be removed and blocking restart

kohsuke committed:
[maven-release-plugin] prepare release jenkins-1.654

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

kohsuke committed:
updated changelog for release

kohsuke committed:
Merge branch 'master' of github.com:jenkinsci/jenkins

jtnord committed:
[FIXED JENKINS-33693] Do not hardcode .bat extension.

As the default maven command is on the path we can not hard code the
command to mvn.bat as since maven 3.something it has been mvn.cmd

As this is on the path we can just use mvn and let Windows do its thing
with PATHEXT which will has both .BAT and .CMD by default.

This fixes ATH failures when the ATH is run on windows.

jglick committed:
Merge pull request #2129 from jglick/immutable-list-JENKINS-33467

[JENKINS-33467] Clarifying that CauseAction.getCauses is immutable

stephenc committed:
Merge pull request #2147 from stephenc/jenkins-33681

[FIXED JENKINS-33681] Plugin filters were failing to be removed and blocking restart

daniel-beck committed:
Update changelog.html

daniel-beck committed:
Update changelog.html

daniel-beck committed:
Revert "Latest & greatest"

This reverts commit 0d0d2a9c133abee1cfd81397c84767638c5a3432.

jtnord committed:
Merge pull request #2148 from jtnord/jenkins-33693

[FIXED JENKINS-33693] Do not hardcode .bat extension.

batmat committed:
Add the .mvn directory and set default Xmx value

Using the new feature introduced in Maven 3.3.1.
See
https://maven.apache.org/docs/3.3.1/release-notes.html#JVM_and_Command_Line_Options
for more details.

Possibly a long term better solution/first encounter for newcomers. When
people can't build Jenkins because of wrong default memory settings.
Like https://goo.gl/Co9kye

The "Xmx800m" was chosen after looking at last successful ci.j.o builds.
Max memory displayed by Maven being ~730MB. So putting 800 to leave the
build some room so that the GC has less chance to have to chime in too
often.

rsandell committed:
Make Fingerprint's XSTREAM field public

Some plugins needs to register compatibility alias etc for
their FingerprintFacets.

rsandell committed:
Make the XStream field be accessible with a getter instead

rsandell committed:
Some javadoc for getXStream

Commit:
Make sure ratings in the changelog works properly on jenkins.io

[FIX WEBSITE-147]

daniel-beck committed:
Merge pull request #2167 from rtyler/good-ratings-are-good

[FIX WEBSITE-147] Make changelog ratings work on jenkins.io

rsandell committed:
Merge pull request #2163 from rsandell/public-fp-xstream

Make Fingerprint's XSTREAM field public

oleg-nenashev committed:
Fix typo in hudson/tools/label.jelly

oleg-nenashev committed:
Merge pull request #2169 from jenkinsci/label-jelly-typo-fix

Fix typo in hudson/tools/label.jelly

jtnord committed:
Merge pull request #2162 from batmat/dot-mvn-dir

Add the .mvn directory and set default Xmx value

daniel-beck committed:
Merge pull request #2155 from daniel-beck/JENKINS-33546-33605

[JENKINS-33546] [JENKINS-33605] Back to Stapler 1.239

jglick committed:
Be quiet.

jglick committed:
Merge pull request #2171 from jglick/noisy-InstallUncaughtExceptionHandler

Be quiet

kohsuke committed:
NioChannelHub must be explicitly shut down

... or else it'll keep running. This can be seen as a large number of zombie threads like the following in the unit test:

"NioChannelHub keys=0 gen=0: Computer.threadPoolForRemoting [#250]" daemon prio=10 tid=0x00007f793d841000 nid=0x3cc1 runnable [0x00007f78e2665000]
   java.lang.Thread.State: RUNNABLE
        at sun.nio.ch.EPollArrayWrapper.epollWait(Native Method)
        at sun.nio.ch.EPollArrayWrapper.poll(EPollArrayWrapper.java:228)
        at sun.nio.ch.EPollSelectorImpl.doSelect(EPollSelectorImpl.java:81)
        at sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:87)
        - locked <0x00000000c50d46d8> (a sun.nio.ch.Util$2)
        - locked <0x00000000c50d46e8> (a java.util.Collections$UnmodifiableSet)
        - locked <0x00000000c50d4690> (a sun.nio.ch.EPollSelectorImpl)
        at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:98)
        at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:102)
        at org.jenkinsci.remoting.nio.NioChannelHub.run(NioChannelHub.java:542)
        at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
        at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:334)
        at java.util.concurrent.FutureTask.run(FutureTask.java:166)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at java.lang.Thread.run(Thread.java:722)

kohsuke committed:
Allow instance methods for @Initializer and @Terminator

This simplifies typical usage of them.

kohsuke committed:
The previous change requires that Guice is setup

... which is done by looking for locating & instantiating ExtensionFinders.

kohsuke committed:
Simplified by using instance methods

kohsuke committed:
Doc improvements

kohsuke committed:
[maven-release-plugin] prepare for next development iteration

kohsuke committed:
[maven-release-plugin] prepare release jenkins-1.655

kohsuke committed:
updated changelog for release

daniel-beck committed:
Noting #2162,#2163,#2171,#2155,#2147,#2148,#2129

daniel-beck committed:
Minor formatting improvements

kzantow committed:
JENKINS-33826 build history badges styling issues

kohsuke committed:
Merge pull request #2176 from jenkinsci/instance-initializer

initializer annotation on instance methods

jglick committed:
Use rerunFailingTestsCount rather than @RandomlyFails.

Commit:
Fix advertised JNLP Port

As far as I can figure, the /TcpSlaveAgentListener/ endpoint headers
tell the slaves what the JNLP port is. When I'm running jenkins behind a
proxy (marathon using the mesos plugin, which bridged networking), even
though I was setting the advertised port per a thread[1], it was not
accurately being returned in the header. This patch fixes that.

[1] https://groups.google.com/d/topic/jenkins-mesos/qUrXpDY07TQ/discussion

Vlatombe committed:
Merge pull request #2189 from yp-engineering/fix_jnlp_advertised_port

Fix advertised JNLP Port

Vlatombe committed:
Update changelog for #2189

jglick committed:
Merge pull request #2188 from jglick/flakiness

Use rerunFailingTestsCount rather than @RandomlyFails

kzantow committed:
Merge pull request #2180 from kzantow/JENKINS-33826-build-history-badge-style-fix

[JENKINS-33826] build history badges styling issues

daniel-beck committed:
Fix link to changelog-old on jenkins.io

lexxxel committed:
Merge remote-tracking branch 'origin/log_failed_login_ip' into log_failed_login_ip

daniel-beck commented:
I don't know wtf is going on here, but… no. Just, no.

Please file a new PR with only the relevant commits (addressing previous review comments) if you want to have this merged.

lexxxel commented:
@daniel-beck I completly forgot about this PR.... what happend here?

daniel-beck commented:
@lexxxel My guess is a merge gone wrong. Best to start over with a new branch.
